File: regia-system/backend/src/bot/whatsapp.js
import axios from "axios";
import pRetry from "p-retry";

const url = (id) => `https://graph.facebook.com/v21.0/${id}/messages`;

async function send(payload) {
  const headers = {
    Authorization: `Bearer ${process.env.WHATSAPP_TOKEN}`,
    "Content-Type": "application/json",
  };

  return pRetry(
    async () => {
      const { data } = await axios.post(
        url(process.env.PHONE_NUMBER_ID),
        payload,
        { headers }
      );
      return data;
    },
    { retries: 3 }
  );
}

export function sendText(to, body) {
  return send({
    messaging_product: "whatsapp",
    to,
    type: "text",
    text: { body },
  });
}

export function sendList(to, header, body, sections) {
  return send({
    messaging_product: "whatsapp",
    to,
    type: "interactive",
    interactive: {
      type: "list",
      header: { type: "text", text: header },
      body: { text: body },
      action: { button: "Ver opções", sections },
    },
  });
}

export function sendButtons(to, body, buttons) {
  return send({
    messaging_product: "whatsapp",
    to,
    type: "interactive",
    interactive: {
      type: "button",
      body: { text: body },
      action: {
        buttons: buttons.map((b) => ({
          type: "reply",
          reply: { id: b.id, title: b.title },
        })),
      },
    },
  });
}

export function sendImage(to, link, caption = "") {
  return send({
    messaging_product: "whatsapp",
    to,
    type: "image",
    image: { link, caption },
  });
}

File: modules/social/whatsapp.py
import os

import requests


def send_whatsapp(phone: str, message: str):
    """
    Envia mensagens via WhatsApp Business API (Meta Cloud).
    """
    token = os.getenv("WABA_TOKEN")
    phone_number_id = os.getenv("WABA_PHONE_ID")
    if not token or not phone_number_id:
        return {"error": "WABA_TOKEN ou WABA_PHONE_ID não configurados."}

    url = f"https://graph.facebook.com/v18.0/{phone_number_id}/messages"

    payload = {
        "messaging_product": "whatsapp",
        "to": phone,
        "type": "text",
        "text": {"body": message},
    }

    headers = {
        "Authorization": f"Bearer {token}",
        "Content-Type": "application/json",
    }

    r = requests.post(url, json=payload, headers=headers)
    return r.json()
